proxy/src/coco/peer.rs

```diff
@@ -113,6 +125,67 @@ pub fn list_projects(peer: &PeerApi) -> Result<Vec<Project>, error::Error> {
         .collect()
 }
 
+/// Get all peer IDs and their branches.
+///
+/// # Errors
+///
+///   * [`error::Error::LibradLock`]
+///   * [`error::Error::Git`]
+pub fn remotes(
+    peer: &PeerApi,
+    owner: &User,
+    project_urn: &RadUrn,
+) -> Result<Vec<Remote>, error::Error> {
+    let project = get_project(peer, project_urn)?;
+    let peer_id = peer.peer_id();
+    let storage = peer.storage();
+    let repo = storage.open_repo(project.urn())?;
+    let refs = repo.rad_refs()?;
+
+    let (local_branches, local_tags) = with_browser(peer, &project.urn(), |browser| {
+        Ok((source::local_branches(browser)?, source::tags(browser)?))
+    })?;
+
+    let owner = owner.to_data().build()?; // TODO(finto): Dirty hack to make our Verified User into a Draft one
+    let mut remotes = vec![Remote {
+        user: owner,
+        branches: local_branches,
+        tags: local_tags,
+    }];
+
+    for remote in refs.remotes.flatten() {
+        let refs = if remote == peer_id {
+            // TODO(finto): Bug in librad
+            continue;
+        } else {
+            let remote_branches = storage.rad_refs_of(&project.urn(), remote.clone())?;
+            remote_branches
+                .heads
+                .keys()
+                .cloned()
+                .map(source::Branch)
+                .collect()
+        };
+
+        // TODO(finto): Can we do this by not going through string?
+        let hash = librad::hash::Hash::hash(remote.to_string().as_bytes());
```

Comment:
@kim: What's the intended way to get the `Hash` of a `PeerId`?

Reply:
Uhm, I think the next `TODO(finto)` answers it -- there is no `RadUrn` which would describe a peer id.

Trying to group remote branches by `User` is not useful -- what if that user has 12 devices, each containing a `master` branch? Instead, the idea would be group the remotes by peer id, then read `rad/self`, and prepend the username to the peer id (where you may consider to truncate the peer id). Note that `rad/self` is only guaranteed to exist if that remote has any branches under `refs/heads` (it can still have things under `rad/`).

To illustrate:

```
cloudhead@hwxyz...
    refs/heads/master
    refs/heads/main
    refs/heads/trunk
cloudhead@hwabc...
    refs/heads/master
    refs/heads/main
    refs/heads/trunk
kim@hwqwer...
    refs/heads/next
anonymous@hwasd...
```

Reply:
Yup, this was to get the handle of the `User`. You're right the next TODO does cover that, didn't backtrack my thought process :sweat_smile: